File: fixtures/cpp/baseline/src/service.cpp
#include "service.h"

namespace example {

std::string Repository::save(const std::string& value) {
    last = value;
    return last;
}

Service::Service(Repository& repository) : repository_(repository) {}

std::string Service::execute(const std::string& name) {
    auto stored = repository_.save(name);
    return std::string(DefaultPrefix) + ":" + stored;
}

Service build_service(Repository& repository) {
    return Service(repository);
}

} // namespace example

File: fixtures/rust/baseline/src/service.rs
pub const DEFAULT_PREFIX: &str = "job";

#[derive(Default)]
pub struct MemoryRepository {
    pub last: String,
}

impl MemoryRepository {
    pub fn save(&mut self, value: &str) -> String {
        self.last = value.to_string();
        value.trim().to_string()
    }
}

pub struct Service {
    repository: MemoryRepository,
}

impl Service {
    pub fn new(repository: MemoryRepository) -> Self {
        Self { repository }
    }

    pub fn execute(mut self, name: &str) -> String {
        let stored = self.repository.save(name);
        format!("{DEFAULT_PREFIX}:{stored}")
    }
}

pub fn build_service(repository: MemoryRepository) -> Service {
    Service::new(repository)
}
